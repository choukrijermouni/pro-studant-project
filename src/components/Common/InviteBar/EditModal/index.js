import {
  TypographyElement,
  Button,
  Status,
  TextArea,
  Col,
  Row,
  neutral,
  TextInput
} from '@pro_boa/ui'
import { createUseStyles } from 'react-jss'
import style from './style'
import Text from './text.json'
import { teams } from 'constants/'
import { useFormik } from 'formik'
import validationSchema from './validator'
import { updateTeamAction } from 'pages/TeamDetails/store'
import { useDispatch, useSelector } from 'react-redux'
import { scrollUp } from 'helpers'

const useStyles = createUseStyles(style)

export default ({ teamName, closeModal, teamIcon, description }) => {
  const {
    modal,
    textArea,
    avatars,
    invitesBox,
    header,
    container,
    flex,
    body,
    overlay,
    disabled,
    titleClass
  } = useStyles()
  const dispatch = useDispatch()
  const { Id } = useSelector(state => state.teamDetails)
  const { handleSubmit, handleChange, resetForm, handleBlur, setFieldValue, touched, errors, values } = useFormik({
    initialValues: {
      name: teamName,
      details: description,
      image: teamIcon
    },
    validate: validationSchema,
    onSubmit: (values) => {
      resetForm()
      scrollUp()
      dispatch(updateTeamAction(Id, values.name, values.details, values.image))
      closeModal()
    },
    enableReinitialize: true
  })
  return (
    <form onSubmit={handleSubmit} data-test='edit-team-modal'>
      <div className={modal}>
        <div className={header}>
          <Col pos='left' className='col' grid={9}>
            <TypographyElement
              component='h3'
              variant='heading3'
              align='left'
              display='flex'
              className={titleClass}
            >
              {Text.title}
            </TypographyElement>
          </Col>
        </div>
        <Row>
          <div className={body}>
            <div className={container}>
              <TypographyElement
                component='h4'
                variant='smallText'
                align='left'
                spacing='0 0 13px'
              >
                {Text.subHeader}
              </TypographyElement>
              <div className={invitesBox}>
                <div className={avatars}>
                  {teams.map((element, id) =>
                    <div key={id} className={flex} data-test='team-icon'>
                      <img id={id} src={element} alt='teamTag' />
                      <div
                        id={element}
                        className={values.image === element ? disabled : overlay}
                        onClick={() => setFieldValue('image', element)}
                      />
                    </div>)}
                </div>
              </div>
              <div>
                <TypographyElement
                  fontSize='13px'
                  lineHeight='16px'
                  color={neutral[6]}
                  spacing='20px 0 13px'
                >
                  {Text.teamName}
                </TypographyElement>
                <TextInput
                  dataTest='team-name-input'
                  ErrorMessageDataTest='team-name-input-error'
                  label=''
                  id='name'
                  name='name'
                  type='text'
                  placeholder={teamName}
                  handleBlur={handleBlur}
                  value={values.name}
                  handleChange={handleChange}
                  status={errors.name && touched.name ? Status.error : null}
                  message={touched.name && errors.name ? errors.name : null}
                />
                <TypographyElement
                  fontSize='13px'
                  lineHeight='16px'
                  color={neutral[6]}
                  spacing='20px 0 13px'
                >
                  {Text.description}
                </TypographyElement>
                <TextArea
                  dataTest='team-description-input'
                  label=''
                  textareaClassName={textArea}
                  id='details'
                  name='details'
                  type='text'
                  placeholder={Text.description}
                  handleBlur={handleBlur}
                  value={values.details}
                  handleChange={handleChange}
                />
              </div>
            </div>
            <Button
              dataTest='save-button'
              type='submit'
              width='100%'
              label={Text.confirmButton}
              handleClick={handleSubmit}
            />
          </div>
        </Row>
      </div>
    </form>
  )
}
