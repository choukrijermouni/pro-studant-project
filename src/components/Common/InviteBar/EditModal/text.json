{
  "title": "Modifier l’équipe",
  "subHeader": "Sélectionnez une icône pour votre équipe",
  "teamName": "Nom de l'équipe ",
  "confirmButton": "Modifier",
  "abortButton": "Annuler",
  "description": "Tapez une description",
  "required": "Champ requis"
}