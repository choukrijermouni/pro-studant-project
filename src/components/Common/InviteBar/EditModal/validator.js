import Text from './text.json'

export default async values => {
  const errors = {}
  if (!values.name) {
    errors.name = Text.required
  }
  return errors
}
