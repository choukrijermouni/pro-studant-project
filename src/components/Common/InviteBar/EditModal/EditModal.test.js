import { shallow, configure } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import renderer from 'react-test-renderer'
import EditModal from '.'
configure({ adapter: new Adapter() })

describe('>>> EditModal --- Shallow Render ManagerCard component', () => {
  let wrapper
  let tree

  beforeEach(() => {
    wrapper = shallow(<EditModal />)
    tree = renderer.create(
      <EditModal />
    ).toJSON()
  })

  it('+++ render correctly and match snapshot', () => {
    expect(tree).toMatchSnapshot()
  })

  it('+++ render the component', () => {
    expect(wrapper).toBeTruthy()
  })
})
