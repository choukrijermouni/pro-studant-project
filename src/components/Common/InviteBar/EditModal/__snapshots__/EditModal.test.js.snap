// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> EditModal --- Shallow Render ManagerCard component +++ render correctly and match snapshot 1`] = `
<div>
  <div
    className="modal-0-2-1"
  >
    <div
      className="header-0-2-2"
    >
      <p
        className="text-0-2-14 text-d0-0-2-15"
      >
        Modifier l’équipe
      </p>
      <img
        alt="notification"
        src={Object {}}
      />
    </div>
    <div>
      <p
        className="text-0-2-16 text-d0-0-2-17"
      >
        Choisissez une icône pour votre équipe
      </p>
      <div
        className="invitesBox-0-2-5"
      >
        <div
          className="avatars-0-2-3"
        >
          <div
            className="overlay-0-2-4"
          />
          <div
            className="flex-0-2-13"
          >
            <img
              alt="teamTag"
              className="teamTag-0-2-11"
              id={0}
              onClick={[Function]}
              src={Object {}}
            />
          </div>
          <div
            className="flex-0-2-13"
          >
            <img
              alt="teamTag"
              className="teamTag-0-2-11"
              id={1}
              onClick={[Function]}
              src={Object {}}
            />
          </div>
          <div
            className="flex-0-2-13"
          >
            <img
              alt="teamTag"
              className="teamTag-0-2-11"
              id={2}
              onClick={[Function]}
              src={Object {}}
            />
          </div>
          <div
            className="flex-0-2-13"
          >
            <img
              alt="teamTag"
              className="teamTag-0-2-11"
              id={3}
              onClick={[Function]}
              src={Object {}}
            />
          </div>
          <div
            className="flex-0-2-13"
          >
            <img
              alt="teamTag"
              className="teamTag-0-2-11"
              id={4}
              onClick={[Function]}
              src={Object {}}
            />
          </div>
          <span
            className="icon-add-plus-alph-iconfont editIcon-0-2-6"
          />
        </div>
      </div>
      <div>
        <p
          className="text-0-2-18 text-d0-0-2-19"
        >
          Nom d'equipe 
        </p>
        <div
          className="inputGroup-0-2-20"
        >
          
          <textarea
            className="input-0-2-9 input-0-2-21 input-d0-0-2-23"
            disabled={false}
            id="1"
            placeholder="input placeholder"
          />
          
        </div>
        <p
          className="text-0-2-24 text-d0-0-2-25"
        >
          Description
        </p>
        <div
          className="inputGroup-0-2-26"
        >
          
          <textarea
            className="textArea-0-2-10 input-0-2-27 input-d0-0-2-29"
            disabled={false}
            id="1"
            placeholder="Description"
          />
          
        </div>
      </div>
      <hr
        className="line-0-2-8"
      />
      <div
        className="licenseButtons-0-2-7"
      >
        <button
          className="button-0-2-30 button-d0-0-2-32"
          onClick={[Function]}
          type="button"
        >
           
          <p
            className="text-0-2-34 text-d0-0-2-35 text-0-2-31 text-d1-0-2-33"
          >
            Annuler
          </p>
        </button>
        <button
          className="button-0-2-36 button-d0-0-2-38"
          onClick={[Function]}
          type="button"
        >
           
          <p
            className="text-0-2-40 text-d0-0-2-41 text-0-2-37 text-d1-0-2-39"
          >
            Sauvgarder
          </p>
        </button>
      </div>
    </div>
  </div>
</div>
`;
